File: admin/src/types/file.ts
export interface UploadedFile {
  bucket: string
  contentType: string
  createdAt: string
  id: string
  originalName: string
  provider: string
  sizeBytes: number
  storageKey: string
  storedName: string
  updatedAt: string
  uploadedById: string
  visibility: 'private' | 'public'
}

export interface UploadFileResponse {
  file: UploadedFile
}

File: api/internal/domain/file/file.go
package file

import "time"

type Visibility string

const (
	VisibilityPrivate Visibility = "private"
	VisibilityPublic  Visibility = "public"
)

type File struct {
	ID           string
	OriginalName string
	StoredName   string
	StorageKey   string
	Bucket       string
	Provider     string
	ContentType  string
	SizeBytes    int64
	Visibility   Visibility
	UploadedByID string
	CreatedAt    time.Time
	UpdatedAt    time.Time
}

func (v Visibility) IsValid() bool {
	return v == VisibilityPrivate || v == VisibilityPublic
}
